File: Exercism Codes/Python/perfect_numbers.py
def classify(number):
    if number <= 0:
        raise ValueError("Classification is only possible for positive integers.")
    aliquot = sum(x for x in range(1, number) if number % x == 0)
    return "abundant" if aliquot > number else ("perfect" if aliquot == number else "deficient")

File: Exercism Codes/JavaScript/perfect-numbers.js
export const classify = (n) => {
    if (n <= 0) throw new Error('Classification is only possible for natural numbers.');
    const aliquot = [...new Array(Math.ceil(n ** 0.5)).keys()]
        .reduce((a, c) => n % c == 0 ? a + c + (c == 1 ? 0 : n / c) : a, 0);
    return aliquot == n ? 'perfect' : aliquot < n ? 'deficient' : 'abundant';
};
